### DISCUSSION TITLE
Add a sha256 for the zipball

### DISCUSSION THREAD

**Discussion by hollyoops** [community/community#42407] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

Could we add the sha256 for a zipball for release or tags API?  Sometimes it's important to get the checksum before download

Ex:

`GET https://api.github.com/repos/hollyoops/RecoilSwift/tags`

Expect to add an SHA-256 to the response

```
[{
"name": "0.2.1",
"zipball_url": "https://api.github.com/repos/hollyoops/RecoilSwift/zipball/refs/tags/0.2.1",
"zipball_sha256": "99e3edfc5ced955da24062996aaf7f98f61c5699",  ================= Expect to add this
"tarball_url": "https://api.github.com/repos/hollyoops/RecoilSwift/tarball/refs/tags/0.2.1",
"tarball_sha256": "xxxxxxx"
"commit": {
       "sha": "44e3edfc5ced955da24062996aaf7f98f61c5635",
       "url": "https://api.github.com/repos/hollyoops/RecoilSwift/commits/44e3edfc5ced955da24062996aaf7f98f61c5635"
 },
"node_id": "MDM6UmVmMzg1MDkzNDYxOnJlZnMvdGFncy8wLjIuMQ=="
}]
```

**Comment by getlost01** [comment]
Yes, you can use the Git tags API to get the SHA-256 checksum of a zipball for a release or tag.

**Reply by hollyoops** [reply]
Could you be more specific? I went through the Git tags API again. I still can not find a way to do this. I do not post a tag for my repos. Just query a zipball sha256 by giving a tag name

**Reply by getlost01** [reply]
To get the SHA-256 checksum for a zipball for a release or tag using the GitHub API, you can include the sha256 field in the list of fields that you want to include in the API response. For example, to get the SHA-256 checksum for the latest release of a repository, you could make a GET request to the `/repos/{owner}/{repo}/releases/latest` endpoint and include `sha256` in the fields parameter like this:
```
GET /repos/{owner}/{repo}/releases/latest?fields=sha256 
```
This will return a JSON object that includes the `sha256` field, which contains the SHA-256 checksum for the zipball for the release.

You can also get the SHA-256 checksum for a zipball for a tag by making a GET request to the `/repos/{owner}/{repo}/zipball/{tag_name}` endpoint and including `sha256` in the `fields` parameter. For example:

```
GET /repos/{owner}/{repo}/zipball/{tag_name}?fields=sha256
```

This will return a JSON object that includes the sha256 field, which contains the `SHA-256 `checksum for the zipball for the tag.

**Reply by hollyoops** [reply]
Thank your reply.
I tried this: `GET https://api.github.com/repos/hollyoops/recoilswift/zipball/0.2.1?fields=sha256`

It's still downloading the zipfile instead of returning a JSON object

**Reply by getlost01** [reply]
Sorry I think, I have less information related to this. Please pardon me for sharing an incomplete solution.

**Comment by hollyoops** [comment]
@getlost01  Looking forward to your solution coming true

**Comment by einenlum** [comment]
It would be indeed nice to get the sha of a zip before download. This would help for caching strategies.
